de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/SarosSessionManager.java

```diff
@@ -532,66 +588,43 @@ public void addResourcesToSession(
-            final List<IResource> resourcesList = mapEntry.getValue();
-
-            // side effect: non shared projects are always partial -.-
-            if (!currentSession.isCompletelyShared(project)) {
-                String projectID = currentSession.getProjectID(project);
-
-                if (projectID == null)
-                    projectID = String.valueOf(SESSION_ID_GENERATOR
-                        .nextInt(Integer.MAX_VALUE));
-
-                currentSession.addSharedResources(project, projectID,
-                    resourcesList);
-
-                projectsToShare.add(project);
-            }
-        }
+        List<IProject> projectsToShare = nextProjectNegotiation
+            .getProjectsToShare(currentSession, SESSION_ID_GENERATOR);
 
         if (projectsToShare.isEmpty()) {
             log.warn("skipping project negotiation because no new projects were added to the current session");
             return;
         }
 
         INegotiationHandler handler = negotiationHandler;
-
         if (handler == null) {
             log.warn("could not start a project negotiation because no handler is installed");
             return;
         }
 
         List<AbstractOutgoingProjectNegotiation> negotiations = new ArrayList<AbstractOutgoingProjectNegotiation>();
 
-        synchronized (this) {
-            if (!startStopSessionLock.tryLock()) {
-                log.warn("could not start a project negotiation because the current session is about to stop");
-                return;
-            }
+        if (!startStopSessionLock.tryLock()) {
+            log.warn("could not start a project negotiation because the current session is about to stop");
+            return;
+        }
 
-            try {
-                for (User user : currentSession.getRemoteUsers()) {
-
-                    TransferType type = TransferType.valueOf(currentSession
-                        .getUserProperties(user).getString(
-                            ProjectNegotiationTypeHook.KEY_TYPE));
-                    AbstractOutgoingProjectNegotiation negotiation = negotiationFactory
-                        .newOutgoingProjectNegotiation(user.getJID(), type,
-                            projectsToShare, this, currentSession);
-
-                    negotiation.setNegotiationListener(negotiationListener);
-                    currentProjectNegotiations.add(negotiation);
-                    negotiations.add(negotiation);
-                }
-            } finally {
-                startStopSessionLock.unlock();
+        try {
+            for (User user : currentSession.getRemoteUsers()) {
+
+                TransferType type = TransferType.valueOf(currentSession
+                    .getUserProperties(user).getString(
+                        ProjectNegotiationTypeHook.KEY_TYPE));
+                AbstractOutgoingProjectNegotiation negotiation = negotiationFactory
+                    .newOutgoingProjectNegotiation(user.getJID(), type,
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "getJID"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=272&issues=AWbfy2AyLpqqSUlcop4G&open=AWbfy2AyLpqqSUlcop4G)

[ISSUE_KEY]: AWbfy2AyLpqqSUlcop4G